InputCombo with readonly option

jackyOO7 opened the pull request:
I added an option readonly to the combobox. This blocks the writing input and a click on the text opens the dropdown to choose from the list. I tested it and for me it works.

It seems removing the InitializeAsDisabled has no effect, but I could have overseen something.

MikeTheWatchGuy commented:
I'm going to bounce this one back.  

I need to make an open announcement about code changes to the core code.

I'll take code, but I'm not going to merge pull requests.

I've had trouble in the past with code changes.
Your code has a syntax error and will crash.  There is also no test harness.

To be fair to you, I've not stated anything in the past about how pulls should be done.
I DO appreciate your help and I will definitely be using your Demo program, which I WILL merge in directly.

It's just the core code base that I'm super picky about.  There are no known bugs and as you may have noticed it's super rare for there to be any.   The only way I feel like I can maintain that level of quality is to do the core code myself.

This is nothing personal... and I sure hope it's not taken that way.

I will be using parts of your code, just not merging it directly.

I will also be adding you to the Readme file as you've done some cool things there.

Does that sound fair?

MikeTheWatchGuy commented:
Hmmm... I can't seem to get it to work.

I can follow the execution path just fine, but when I added it to the Demo_All_Widgets code, it doesn't keep me from choosing something.
What happens when someone sets both disabled and readonly?

MikeTheWatchGuy commented:
I'm going to close this... I wish there was a way of indicating on GitHub that you suggested these changes.  I've completed making them with some modifications.  

Please keep finding ways to improve it... I'm listening...

jackyOO7 commented:
Hey Mike
Sorry I was busy this weekend. We had nice weather so I spent most of the time outside and not in front of my PC ;)
I did not want to break your code or bungle up your workflow. I needed the readonly function for my project, so I fixed it (for me). Then I feared it would be overwritten with the next update of PySimpleGUI. As I really don't know yet how this GitHub thing works, I just forked and changed and submitted a pull request to get your attention. I didn't find a way to contact you directly.
I thought I'd help you more if I fixed it myself and sent it for you to inspect and eventually merge than just putting up an issue request for it. Just tell me how you like to handle things in the future.

Keep up the good work!

PS: are you into watches or is that just a name? ;)

MikeTheWatchGuy commented:
Shoot me an email in the future.  We can work together on new features so your work doesn't get tossed if I don't use it directly but instead use it as a template to get it done.  Some people may be turned off by the unusual process I have at the moment where I'm the only one making changes to the core code.

I most certainly was thankful to see your stuff!  Keep it coming.

If you want to contact me in the future, shoot me an email to mike@pysimplegui.org and it'll get to me.

Yea, was into watches... less so now that I've got this and another big project I'm working on.

www.GruenWristwatches.com is where I was posting my stuff.  I am really into that brand because it used  to be a huge popular brand and now all of the history has been lost.  I've done all I can to preserve what I've found.
